main/src/ca/uwaterloo/flix/language/phase/EntryPoints.scala

```diff
@@ -119,6 +119,10 @@ object EntryPoints {
   private def isTest(defn: TypedAst.Def): Boolean =
     defn.spec.ann.isTest
 
+  /** Returns `true` if `defn` has the Assert effect. */
+  private def hasAssert(defn: TypedAst.Def): Boolean =
+    defn.spec.eff.effects.contains(Symbol.Assert)
```

Comment:
This is a syntactic check. Things might go wrong if I have a function `foo(): Unit \ Assert - Assert`

Reply:
We should at least add a comment if this is acceptable.